amr-wind/wind_energy/actuator/disk/disk_spreading.H

```diff
@@ -0,0 +1,170 @@
+#ifndef DISK_SPREADING_H_
+#define DISK_SPREADING_H_
+
+#include "amr-wind/wind_energy/actuator/actuator_utils.H"
+#include "amr-wind/core/FieldRepo.H"
+#include <stdexcept>
+
+namespace amr_wind {
+namespace actuator {
+namespace ops {
+
+/**
+ * @brief  A collection of spreading functions
+ * This class allows for polymorphic spreading functions.
+ * The concept is based on Impossibly Fast C++ Delegates proposed here:
+ * https://www.codeproject.com/Articles/11015/The-Impossibly-Fast-C-Delegates
+ *
+ * @tparam T
+ */
+template <typename T>
+class SpreadingFunction
+{
+public:
+    void operator()(
+        T* actObj,
+        const int lev,
+        const amrex::MFIter& mfi,
+        const amrex::Geometry& geom)
+    {
+        (this->*m_function)(actObj, lev, mfi, geom);
+    }
+
+    SpreadingFunction(const SpreadingFunction&) = delete;
+    void operator=(const SpreadingFunction&) = delete;
+
+    void (SpreadingFunction::*m_function)(
+        T* actObj, const int, const amrex::MFIter&, const amrex::Geometry&);
```

Comment:
Is a virtual function call much slower than a call to via a function pointer?

Reply:
probably not

Reply:
Thanks, so we probably could have done without this complexity?

Reply:
probably. if you have a suggestion or requested change let me know.  There are several ways to do a design. The actual interface for this is not really any more complicated than a virtual function call. It could be done either way.  Off the top of my head, with a virtual call I could could use a unique_ptr with some logic to select the right object type based on the string handle I'm using to select the function in `initialize()`.  I don't see how that is much different than this. Not sure if it is worth a re-write at this point.

If you are going to just ask me leading questions to criticize the code then I will remove you as a reviewer. If you have an actual suggestion to make the code simpler and cleaner I'm all ears.